internal/export/distro/registrations.go

```diff
@@ -163,21 +164,41 @@ func (reg *registrationInfo) update(newReg contract.Registration) bool {
 	return true
 }
 
-func (reg registrationInfo) processEvent(event *models.Event) {
-	// Valid Event Filter, needed?
+func (reg registrationInfo) processMessage(msg msgTypes.MessageEnvelope) {
```

Comment:
This method (and others) should be able to confirm the message contains an `event` payload and not simply `application/json` or `application/cbor` type. It seems to me that the `msgTypes.MessageEnvelope` need not be so opaque. Are there plans to add a property to the envelope at a later time?

Reply:
We can discuss, but I don't think the intent has ever been to make the MessageEnvelope aware of what it contains. In theory it could contain anything, our only current use case is events.

Reply:
Sounds good. Yes, since it is only use case I agree this is beyond the scope of this issue.